homeassistant/components/abode/__init__.py

```diff
@@ -110,18 +110,34 @@ async def async_setup_entry(hass, config_entry):
     username = config_entry.data.get(CONF_USERNAME)
     password = config_entry.data.get(CONF_PASSWORD)
     polling = config_entry.data.get(CONF_POLLING)
+    cache = hass.config.path(DEFAULT_CACHEDB)
+
+    # For previous config entries where unique_id is None
+    if config_entry.unique_id is None:
+        hass.config_entries.async_update_entry(
+            config_entry, unique_id=config_entry.data[CONF_USERNAME]
```

Comment:
I'm assuming `unique_id` for the config entry only has to be unique within the integration itself (i.e. unique to other Abode config entries) and all integrations. The username is the user's email address which could obviously be used for other integration logins.

Reply:
Yes.